presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoContainerRemoteFunction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestPrestoContainerRemoteFunction
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected ContainerQueryRunner createQueryRunner()
+            throws Exception
+    {
+        return new ContainerQueryRunner(
+                ContainerQueryRunner.DEFAULT_COORDINATOR_PORT,
+                ContainerQueryRunner.TPCH_CATALOG,
+                ContainerQueryRunner.TINY_SCHEMA,
+                ContainerQueryRunner.DEFAULT_NUMBER_OF_WORKERS,
+                ContainerQueryRunner.DEFAULT_FUNCTION_SERVER_PORT,
+                true);
+    }
+
+    @Test
+    public void testRemoteBasicTests()
+    {
+        // The server used here is the executable provided by presto-function-server.
```

Comment:
Didn't follow this... What is the code-path from the co-ordinator ? We shouldn't allow that in native worker.